src/main/java/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -1465,13 +1417,74 @@ private void parseItemset(QuestionDef q, Element e) {
+        String labelXpath = child.getAttributeValue("", REF_ATTR);
+        boolean labelItext = false;
+
+        if (XFormUtils.showUnusedAttributeWarning(child, labelUA)) {
+            reporter.warning(XFormParserReporter.TYPE_UNKNOWN_MARKUP, XFormUtils.unusedAttWarning(child, labelUA), getVagueLocation(child));
+        }
+
+        if (labelXpath != null) {
+            if (labelXpath.startsWith("jr:itext(") && labelXpath.endsWith(")")) {
+                labelXpath = labelXpath.substring("jr:itext(".length(), labelXpath.indexOf(")"));
+                labelItext = true;
+            }
+        } else {
+            throw new XFormParseException("<label> in <itemset> requires 'ref'");
+        }
+
+        XPathPathExpr labelPath = XPathReference.getPathExpr(labelXpath);
+        itemset.labelRef = FormInstance.unpackReference(getAbsRef(new XPathReference(labelPath), itemset.nodesetRef));
+        itemset.labelExpr = new XPathConditional(labelPath);
+        itemset.labelIsItext = labelItext;
+    }
+
+    private void parseItemsetCopyElement(Element child, ItemsetBinding itemset, Vector<String> copyUA) {
+        String copyRef = child.getAttributeValue("", REF_ATTR);
+        if (XFormUtils.showUnusedAttributeWarning(child, copyUA)) {
+            reporter.warning(XFormParserReporter.TYPE_UNKNOWN_MARKUP, XFormUtils.unusedAttWarning(child, copyUA), getVagueLocation(child));
+        }
+        if (copyRef == null) {
+            throw new XFormParseException("<copy> in <itemset> requires 'ref'");
+        }
+        itemset.copyRef = FormInstance.unpackReference(getAbsRef(new XPathReference(copyRef), itemset.nodesetRef));
+        itemset.copyMode = true;
+    }
+
+    private void parseItemsetValueElement(Element child, ItemsetBinding itemset, Vector<String> valueUA) {
+        String valueXpath = child.getAttributeValue("", REF_ATTR);
+
+        if (XFormUtils.showUnusedAttributeWarning(child, valueUA)) {
+            reporter.warning(XFormParserReporter.TYPE_UNKNOWN_MARKUP, XFormUtils.unusedAttWarning(child, valueUA), getVagueLocation(child));
+        }
+        if (valueXpath == null) {
+            throw new XFormParseException("<value> in <itemset> requires 'ref'");
+        }
+
+        XPathPathExpr valuePath = XPathReference.getPathExpr(valueXpath);
+        itemset.valueRef = FormInstance.unpackReference(getAbsRef(new XPathReference(valuePath), itemset.nodesetRef));
+        itemset.valueExpr = new XPathConditional(valuePath);
+        itemset.copyMode = false;
+    }
+
+    private void parseItemsetSortElement(Element child, ItemsetBinding itemset) {
+        String sortXpathString = child.getAttributeValue("", REF_ATTR);
+        if (sortXpathString == null) {
+            throw new XFormParseException("<sort> in <itemset> requires 'ref'");
+        }
+
+        XPathPathExpr sortPath = XPathReference.getPathExpr(sortXpathString);

```

Comment:
Is it easy enough to make this an XPath expression, rather than a reference? The other two are references because that's part of the XForms specification, but there's no specific reason that we have to be that restrictive.

If it adds a lot of work to make this an expression rather than a path references, no worries for now, just seemed like a convenient capacity to add if easy.